Documentation and bug fixes

gkuenning opened the pull request:
This PR contains four fixes:

1) Add a minimal .gitignore
2) Fix some documentation errors
3) Fix some crashes that happen when a field is None
4) Fix a bug that put stray spaces into the getBranches() result

gkuenning committed:
Add a minimal .gitignore

gkuenning committed:
Fix several small documentation errors

gkuenning committed:
Fix several places that could cause crashes

Fix two places that would crash if a variable was None, plus one where
the None check was done on the wrong variable.

gkuenning committed:
Don't double-stringify the branch list

ParseRcs was turning the branches list into a string before calling
setBranches; setBranches then treated that string as a list and
re-stringified it.  The result was that a branch list that contained
1.1.1.1 and 1.2.1.1 would become '1 . 1 . 1 . 1   1 . 2 . 1 . 1'
rather than the correct '1.1.1.1 1.2.1.1'.

This is technically a functionality change, but presumably there is no
existing code that depends on the incorrect behavior.

ben-cohen commented on src/editrcs/__init__.py:
```diff
@@ -809,7 +811,7 @@ def getStrict(self, to_bool = True):
         False)."""
         if to_bool:
-            if value not in [None, ""]:
+            if self.__strict not in [None, ""]:
```
Good catch, thank you!  I see that the last sentence in the comment above should not be there.

I don't mind if you want to remove it in your pull request, otherwise I can do it later.

ben-cohen replied:
Done in commit b585dc9311d.

ben-cohen commented on src/editrcs/__init__.py:
```diff
@@ -65,6 +65,8 @@ def AddAtQuoting(s):
 def RemoveAtQuoting(s):
     """Return the string s with RCS at quoting removed."""
+    if s is None:
+        return ''
```
I think it should return `None` and not `''` here.  Otherwise, for example, the call to `getDesc()` in `validate()` will never fail.

Do you agree?

ben-cohen replied:
```suggestion
        return None
```

ben-cohen commented on src/editrcs/__init__.py:
```diff
@@ -1062,7 +1062,7 @@ def ParseRcs(text):
                 break
             branches_list.append(t)
-        delta.setBranches(" ".join(branches_list))
+        delta.setBranches(branches_list)
```
Thank you for your helpful commit message!

ben-cohen reviewed (commented):
Thank you for all these fixes!

I have suggested a change in `RemoveAtQuoting()`.  Everything else looks good to me.

gkuenning commented:
> I don't mind if you want to remove it in your pull request, 
> otherwise
> I can do it later.

It would be simpler for me if you could do it.

>  def RemoveAtQuoting(s):
>      """Return the string s with RCS at quoting removed."""
> +    if s is None:
> +        return ''
>
> I think it should return `None` and not `''` here.  Otherwise, 
> for example, the call to `getDesc()` in `validate()` will never 
> fail.
>
> Do you agree?

Yes, that makes sense.
-- 
    Geoff Kuenning   ***@***.*** 
    https://www.cs.hmc.edu/~geoff/

Statistics don't bore people, people bore people.

ben-cohen reviewed (approved):
Thank you again for all your fixes.

Looks good to me! :tada:

ben-cohen committed:
Preserve None passed as RemoveAtQuoting argument

If the argument to `RemoveAtQuoting()` is `None` then it should return `None`.  Otherwise, for example, the call to `getDesc()` in `validate()` will never fail.
